turbopy/core.py

```diff
@@ -942,7 +937,13 @@ def initialize(self):
 
         This gets called once before the main simulation loop.
         """
-        pass
+        if "directory" in self._input_data:
+            d = Path(self._input_data["directory"])
+        else:
+            # Set a default output directory
+            d = Path("default_output")
+            self._input_data["directory"] = str(d)
```

Comment:
I don't think we need this check here. If the diagnostic is set up through the `read_diagnostics_from_input` function, then the default value for the "directory" is already set. 

While this could give a `KeyNotFoundError` if the  "directory" is not set, I think it would be a simple enough error to find and so should be left to the end users. I prefer having the code give an error to having the default output folder set in two different places.